From 4cec0c0cc03d8e9e942be6126676853603487575 Mon Sep 17 00:00:00 2001
From: babak sarashki <babak.sarashki@windriver.com>
Date: Sun, 3 Nov 2019 15:25:21 -0800
Subject: [PATCH 19/19] openldap fedora systemd

From stx 1901: openldap-fedora-systemd.patch
Skip any empty parameters when parsing command line options.
This is required because systemd does not expand variables the same way as shell does,
we need it because of an empty SLAPD_OPTIONS in environment file.

Fedora specific patch.

Author: Jan Vcelak <jvcelak@redhat.com>
---
 servers/slapd/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/servers/slapd/main.c b/servers/slapd/main.c
index c212209..23f7656 100644
--- a/servers/slapd/main.c
+++ b/servers/slapd/main.c
@@ -685,6 +685,10 @@ unhandled_option:;
 		}
 	}
 
+	/* skip empty parameters */
+	while ( optind < argc && *argv[optind] == '\0' )
+		optind += 1;
+
 	if ( optind != argc )
 		goto unhandled_option;
 
-- 
2.17.1

